app/src/main/java/com/duckduckgo/app/browser/BrowserTabViewModel.kt

```diff
@@ -1398,20 +1423,49 @@ class BrowserTabViewModel(
         }
     }
 
-    fun onAddFavoriteMenuClicked() {
+    private suspend fun editSiteBookmark(url: String) {
+        withContext(dispatchers.io()) {
+            bookmarksState.value?.firstOrNull { it.url == url }
```

Comment:
why do we need to change the value of `LiveData ` here? Nothing has changed until the user interacts with the `Dialog`

Reply:
@malmstein curious about this one. I don't think this line is changing the value of `LiveData` right? I think is just using the latest value stored (and emitted) as a cache to perform a query.

Reply:
I can see how it may be confusing and will extract to another variable to make the code more clear.